packages/contracts/contracts/Tally.sol

```diff
@@ -113,6 +105,23 @@ contract Tally is Clone, SnarkCommon, Hasher, DomainObjs, ITally {
     }
   }
 
+  /// @notice Get the tally results for a specific index
+  /// @param index The index to query
+  /// @return The tally result
+  function getTallyResults(uint256 index) public view returns (TallyResult memory) {
+    return tallyResults[index];
+  }
+
+  /// @notice Get complete tally results
+  /// @return The tally results
+  function getCompleteTallyResults() public view returns (TallyResult[] memory) {
+    TallyResult[] memory results = new TallyResult[](totalTallyResults);
+    for (uint256 i = 0; i < totalTallyResults; i++) {
+      results[i] = tallyResults[i];
+    }
+    return results;
+  }
```

Comment:
Copying all the tally results into a memory is not a good solution.
If you need results for specific options, you can use `getTallyResults` with specified indices or you can use subgraph.